seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/filter/FilterFieldTransform.java

```diff
@@ -35,20 +35,25 @@
 import lombok.extern.slf4j.Slf4j;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Set;
 import java.util.stream.Collectors;
 
 @Slf4j
 public class FilterFieldTransform extends AbstractCatalogSupportTransform {
     public static final String PLUGIN_NAME = "Filter";
     private int[] inputValueIndex;
+    private Set<Integer> inputValueIndexSet;
```

Comment:
Can we keep only one variable? Depending on the mode, the value in the variable is kept or deleted.

Reply:
sure, the redundant variable has been removed, please take a look again.